### PR TITLE
some minor cleanup + use selector

### PR DESCRIPTION
**Pull request by braver**

### CODE REVIEW

File: linter.py [hunk 1]
```diff
@@ -52,10 +36,10 @@ def find_gopaths(self):
             if p not in goroot and p not in gopath:
                 gopath.append(p)
 
-        if persist.debug_mode():
-            persist.printf("{}: {} {}".format(self.name,
-                                              os.path.basename(self.filename or '<unsaved>'),
-                                              "guessed GOPATH=" + os.pathsep.join(gopath)))
+        persist.debug("{}: {} {}".format(
```
Comment (kaste) [inline linter.py]: `persist.debug` is deprecated.

Define ```logger = logging.getLogger('SublimeLinter.plugin.golint')``` at the top. And the just use `logger.info(...)`

Comment (braver) [reply linter.py]: Yeah I was wondering about that and couldn't find a good example. Thanks.

File: linter.py [hunk 2]
```diff
@@ -64,9 +48,9 @@ def run(self, cmd, code):
...
-        if persist.debug_mode():
-            persist.printf('{}: {} {}'.format(self.name,
-                                              os.path.basename(self.filename or '<unsaved>'),
-                                              cmd or '<builtin>'))
+        persist.debug('{}: {} {}'.format(
+            self.name,
+            os.path.basename(self.filename or '<unsaved>'),
+            cmd or '<builtin>'))
 
         return self.tmpfile(cmd, code, suffix=self.get_tempfile_suffix())
```
Comment (kaste) [inline linter.py]: Instead `return self.tmpfile(cmd, code)`

**Comment by braver** [comment]
Good point. Probably not? I don't even know what gopaths are.

**Comment by kaste** [comment]
With actual users this would be easier. 😸 Probably

```
	'env': {'GOPATH': '$folder'}
```

would be the 90% use case solved. But we don't know, so it has to stay.

**Comment by kaste** [comment]
So after the discussion in gotype it is clear that we can remove `find_gopaths`

**Pull request merged**